File: be/src/statistics/dto/column-statistics.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class ColumnStatisticsDto {
  @ApiProperty({ example: 'uuid-here', description: 'ID của cột' })
  columnId: string;

  @ApiProperty({ example: 'To Do', description: 'Tên cột' })
  columnName: string;

  @ApiProperty({ example: 10, description: 'Tổng số nhiệm vụ trong cột' })
  totalTasks: number;

  @ApiProperty({ example: 5, description: 'Số nhiệm vụ đã hoàn thành' })
  doneTasks: number;

  @ApiProperty({ example: 5, description: 'Số nhiệm vụ chưa hoàn thành' })
  todoTasks: number;

  @ApiProperty({ example: 50.0, description: 'Phần trăm tiến độ' })
  progress: number;
}

File: backend-java/src/main/java/com/qlda/backend/statistics/dto/ColumnStatisticsDto.java
package com.qlda.backend.statistics.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ColumnStatisticsDto {
    private String columnId;
    private String columnName;
    private Integer totalTasks;
    private Integer doneTasks;
    private Integer todoTasks;
    private Double progress;
}

